modules/data-streams/src/test/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleFixtures.java

```diff
@@ -148,12 +148,12 @@ static DataStreamLifecycle.Template randomDataLifecycleTemplate() {
         ResettableValue<List<DataStreamLifecycle.DownsamplingRound>> downsampling = randomResettable(
             DataStreamLifecycleFixtures::randomDownsamplingRounds
         );
-        return DataStreamLifecycle.createDataLifecycleTemplate(
-            frequently(),
-            randomResettable(ESTestCase::randomTimeValue),
-            downsampling,
-            randomResettable(() -> randomSamplingMethod(downsampling.get()))
-        );
+        return DataStreamLifecycle.dataLifecycleBuilder()
+            .enabled(frequently())
+            .dataRetention(randomResettable(ESTestCase::randomTimeValue))
+            .downsamplingRounds(downsampling)
+            .downsamplingMethod(randomResettable(() -> randomSamplingMethod(downsampling.get())))
+            .buildTemplate();
```

Comment:
There are one or two similar usages where we do actually lose a bit of value by using a builder. In these cases, it would actually be nice to get a compilation error if you add a new field but forget to add it to these methods that generate random instances. That way, you maintain full coverage by having to specify a random value for the new field here as well.

However, because `DataStreamLifecycle.LifecycleType` is package private, we can't currently use the constructor of `DataStreamLifecycle`. The only options I could think of are using a builder (like I currently do in this PR) or make the `LifecycleType` enum public. Thoughts are welcome :)